Cython/Shadow.py

```diff
@@ -565,7 +565,10 @@ def __init__(self, module):
     def __getattr__(self, item):
         if item.startswith('__') and item.endswith('__'):
             raise AttributeError(item)
-        return __import__(item)
+        try:
+            return __import__(item)
+        except ModuleNotFoundError:
+            raise AttributeError(item) from None
```

Comment:
`ModuleNotFoundError` is python3.6+, before that it was `ImportError`. The harder problem is the `from None` which is not available in python2.7

Reply:
Much as I don't like it, something involving `exec` would probably work here (given the syntax is genuinely incompatible):

Something like:

```
def __getattr__(self):
     # Python 2 version, possibly not fully solving the problem

if sys.version_info >= (3, 6):
    __getattr__ = exec("... full version as in your code...")
elif sys.version_info > (2, 7):
    __getattr__ = exec("... work round ModuleNotFoundError...")
```

Reply:
I think I found a solution without exec